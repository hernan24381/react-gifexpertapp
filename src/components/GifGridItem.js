import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

    //console.log({img});

    return (
        <div className="card animate__animated animate__fadeOut">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}



GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,

}
