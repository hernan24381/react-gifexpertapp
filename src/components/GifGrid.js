import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
//import { getGifs } from '../helpers/getGifs';

import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

   /*  const [images, setImages] = useState([]);

    useEffect(() => {
        //GetGifs(category).then(imgs => setImages(imgs));
        getGifs(category).then(setImages);  // es lo mismo que la anterior linea
    }, [category]) */

    const {data:images, loading} = useFetchGifs( category );
    //console.log(loading);
    //console.log(data);
  
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Loading...</p>}

            <div className="card-grid">
                <ol>
                    {
                        images.map( img => (
                            <GifGridItem 
                                key={img.id}
                                {...img} 
                            />
                        ))
                    }
                </ol>
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
