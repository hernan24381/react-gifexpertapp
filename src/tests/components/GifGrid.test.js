import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas de <GifGrid />', () => {
    
    const category = 'samurai x';
    
    test('debe hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();

    })
    
    test('debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: '123',
            url: 'http://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    



})




