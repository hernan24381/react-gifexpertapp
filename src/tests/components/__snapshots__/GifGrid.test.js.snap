// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    samurai x
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cualquier cosa"
        url="http://localhost/cualquier/cosa.jpg"
      />
      <GifGridItem
        id="123"
        key="123"
        title="Cualquier cosa"
        url="http://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas de <GifGrid /> debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    samurai x
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
